### PR TITLE
txbuilder: add setLockTime

### PR DESCRIPTION
**Pull request by dcousens**
WIP, needs tests.

Open to feedback, not the biggest fan, but, its simple enough.

Maybe I should use a property instead?

### CODE REVIEW

**Comment by dcousens** [comment]
Resolves https://github.com/bitcoinjs/bitcoinjs-lib/issues/437

**Comment by dcousens** [comment]
ping @jprichardson thoughts on API?

**Comment by jprichardson** [comment]
`setLockTime()` is fine. 

An aside, builders are commonly used in conjunction with a fluent interface. Since it doesn't, what's the rationalization for splitting up `Transaction` / `TransactionBuilder`? (Really doesn't have to be answered here)

**Comment by dcousens** [comment]
`Transaction` is just the data structure,  no state,  capable of being manually mutated and it will still serialize correctly.
To which, yes, it probably does make sense to remove `addOutput` on it and just have `transaction.outputs.push`  etc.

**Comment by dcousens** [comment]
> An aside, builders are commonly used in conjunction with a fluent interface. 

Any ideas on how we could improve this interface?

File: src/transaction_builder.js [hunk 1]
```diff
@@ -135,6 +137,21 @@ function TransactionBuilder (network) {
...
+TransactionBuilder.setLockTime = function (locktime) {
+  typeforce(types.Uint32, locktime)
+
+  // if any signatures exist, throw
+  if (this.inputs.some(function (input) {
+    if (!input.signatures) return false
+
+    return input.signatures.some(function (s) { return s })
+  })) {
+    throw new Error('No, this would invalidate signatures')
```
Comment (weilu) [inline src/transaction_builder.js]: I still find this error message funny because it answers "no" without any question asked :P

**Comment by weilu** [comment]
The API looks ok to me.

File: src/transaction_builder.js (hunk 1 above)
Comment (dcousens) [inline src/transaction_builder.js]: Haha, I know.  But, it works.
If you can think of something for `3.0.0`, go for it :)

**Comment by dcousens** [comment]
@jprichardson not going to merge this without status checks as tests have been added.
Please verify manually and merge when ready. :+1:

**Comment by dcousens** [comment]
Needed https://github.com/bitcoinjs/bitcoinjs-lib/pull/524,  so merged manually after local verification.
@jprichardson please review, verify and merge.

**Comment by jprichardson** [comment]
:+1:

**Pull request merged**